persistence/jdbc-common/src/main/java/org/infinispan/persistence/jdbc/common/impl/BaseJdbcStore.java

```diff
@@ -108,7 +108,8 @@ public CompletionStage<Boolean> isAvailable() {
          try {
             connection = connectionFactory.getConnection();
             return connection != null && connection.isValid(10);
-         } catch (SQLException e) {
+         } catch (Throwable t) {
+            log.storeAvailabilityException(t);
             return false;
```

Comment:
wouldn't be better to throw the exception and let the `PersistenceManagerImpl` do the logging? 
```suggestion
            throw CompletableFutures.asCompletionException(t);
```

Reply:
Good call, comment addressed and I've also updated the logging in `PersistenceManagerImpl` so that a WARN log occurs on availability changed. Debug logging didn't seem sufficient to me given that a loss of persistence can bring an entire cache down.